packages/concerto-core/lib/introspect/property.js

```diff
@@ -178,15 +179,15 @@ class Property extends Decorated {
 
         const parent = this.getParent();
         if(!parent) {
-            throw new Error('Property ' + this.name + ' does not have a parent.');
+            throw new IllegalModelException('Property ' + this.name + ' does not have a parent.');
         }
         const modelFile = parent.getModelFile();
         if(!modelFile) {
-            throw new Error('Parent of property ' + this.name + ' does not have a ModelFile!');
+            throw new IllegalModelException('Parent of property ' + this.name + ' does not have a ModelFile!');
         }
         const result = modelFile.getFullyQualifiedTypeName(this.type);
         if(!result) {
-            throw new Error('Failed to find fully qualified type name for property ' + this.name + ' with type ' + this.type );
+            throw new TypeNotFoundException(this.type,'Failed to find fully qualified type name for property ' + this.name + ' with type ' + this.type);
```

Comment:
Here the error code should be `TypeNotFoundException`.

Reply:
> Here the error code should be `TypeNotFoundException`.

But I can see the default error type is defined as `ErrorCodes.TYPE_NOT_FOUND_EXCEPTION`.
Should I pass the error code again while throwing from from the `if statement` ?

Reply:
We should add it explicitly just in case default changes.

But the error above also doesn't add error code.

Reply:
Sorry you are correct